environment/src/test/scala/ru/primetalk/funtik/environment/generator/BSPTreeSpec.scala

```diff
@@ -0,0 +1,25 @@
+package ru.primetalk.funtik.environment.generator
+
+import org.specs2.Specification
+import org.specs2.matcher.MatchResult
+import ru.primetalk.funtik.environment.geom2d.Geom2dUtils.Rectangle
+import ru.primetalk.funtik.environment.genereator._
+import ru.primetalk.funtik.environment.genereator.utils.Random
+import ru.primetalk.funtik.environment.geom2d.Geom2dUtils
+
+class BSPTreeSpec extends Specification { def is = s2"""
+
+  BSPTree should
+    generate some rooms    $generate
+
+  """
+  def bspTree = new BSPTree
+
+  def generate: MatchResult[Tree[Geom2dUtils.Rectangle]] = {
+    val rect = Rectangle(0 -> 0, 100 -> 100)
+    val stream = Random.stream(10)
+    val result = bspTree.generate(rect).runA(stream).value
+    result must haveClass[Node[Rectangle]]
```

Comment:
Is this a real test?

Reply:
Yes, why not? It's a simplest test which only checks that method doesn't fails and return value of valid type. But I understand that "tests" are not exhaustive.